File: 3148-maximum-difference-score-in-a-grid/3148-maximum-difference-score-in-a-grid.cpp
class Solution {
public:
    int maxScore(vector<vector<int>>& grid) {
        int res = INT_MIN, n = grid.size(), m = grid[0].size();
        for(int i=0; i<n; i++) {
            for(int j=0; j<m; j++) {
                int mx = min(i > 0 ? grid[i-1][j] : 1e6, j > 0 ? grid[i][j-1] : 1e6);
                res = max(res, grid[i][j] - mx);
                grid[i][j] = min(grid[i][j], mx);
            }
        }
        return res;
    }
};

File: 3148-maximum-difference-score-in-a-grid/3148-maximum-difference-score-in-a-grid.java
class Solution {
    public int maxScore(List<List<Integer>> grid) {
        int res = Integer.MIN_VALUE, n = grid.size(), m = grid.get(0).size();
        for(int i=0; i<grid.size(); i++) {
            for(int j=0; j<grid.get(0).size(); j++) {
                int mn = Math.min(i > 0 ? grid.get(i-1).get(j) : 1000_000, j > 0 ? grid.get(i).get(j-1) : 1000_000);
                res = Math.max(res, grid.get(i).get(j) - mn);
                grid.get(i).set(j, Math.min(mn, grid.get(i).get(j)));
            }
        }
        return res;
    }
}